File: web/src/api/pas/performanceReviewItemUser.js
import service from '@/utils/request'

export const getLastPRICreatePRIU = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/getLastPRICreatePRIU",
        method: 'post',
        data
    })
}

export const getPRIUByPRIID = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/getPRIUByPRIID",
        method: 'post',
        data
    })
}
export const updatePRIU = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/updatePRIU",
        method: 'put',
        data
    })
}

export const removePRIU = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/removePRIU",
        method: 'delete',
        data
    })
}

export const createPRIU = (data) => {
   return service({
       url: "/PerformanceReviewItemUser/createPRIU",
       method: 'post',
       data
   })
}
export const getPRIUListByUser = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/getPRIUListByUser",
        method: 'post',
        data
    })
}
export const updatePRIUStatusByPRIID = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/updatePRIUStatusByPRIID",
        method: 'post',
        data
    })
}

export const getPRIUListByStatus = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/getPRIUListByStatus",
        method: 'post',
        data
    })
}

export const updatePRIUStatustByIds = (data) => {
    return service({
        url: "/PerformanceReviewItemUser/updatePRIUStatustByIds",
        method: 'post',
        data
    })
}

File: server/router/pas/performance_review_item_user.go
package pas

import (
	"gin-vue-admin/api/v1/pas"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitPerformanceReviewItemUserRouter(Router *gin.RouterGroup) {
	PerformanceReviewItemUserRouter := Router.Group("PerformanceReviewItemUser").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		PerformanceReviewItemUserRouter.POST("getLastPRICreatePRIU", pas.GetLastPRICreatePRIU)
		PerformanceReviewItemUserRouter.POST("getPRIUByPRIID", pas.GetPRIUByPRIID) 
		PerformanceReviewItemUserRouter.PUT("updatePRIU", pas.UpdatePRIU)  
		PerformanceReviewItemUserRouter.DELETE("removePRIU", pas.RemovePRIU)  
		PerformanceReviewItemUserRouter.POST("createPRIU", pas.CreatePRIU)   
		PerformanceReviewItemUserRouter.POST("getPRIUListByUser", pas.GetPRIUListByUser)
		PerformanceReviewItemUserRouter.POST("updatePRIUStatusByPRIID", pas.UpdatePRIUStatusByPRIID)
		PerformanceReviewItemUserRouter.POST("getPRIUListByStatus", pas.GetPRIUListByStatus)
		PerformanceReviewItemUserRouter.POST("updatePRIUStatustByIds", pas.UpdatePRIUStatusByIds)
		
	}
}
